pyscriptjs/src/interpreter_client.ts

```diff
@@ -61,4 +72,16 @@ export class InterpreterClient extends Object {
         }
         return result;
     }
+
+    async pyimport(mod_name: string): Promise<PyProxy> {
+        return await this._remote.pyimport(mod_name);
```

Comment:
`async` methods are already a *Promise* hence it's never needed to `return await` ... just `return`. The difference is that here we have 2 ticks (microtasks) instead of one. I would personally drop that `await` in there.

Reply:
Thanks, fixed in https://github.com/pyscript/pyscript/pull/1258/commits/b4d138e6718caa13ef38fb9afd6729729be29c20

Reply:
It's not the same because the thing awaited here is a task not a promise. So desugaring a bit with await does eager evaluation, like writing:
```js
return f(res);
```
Where without await is like writing:
```js
return { then(res){ return f(res); } };
```

I think a happy medium would be:
```js
return task.schedule_async();
```
I think this is actually logically equivalent to the original but it drops the tick.

Reply:
what I meant is that `async` is already a promise and anything returned as *awaitable* doesn't need `await` upfront ... actually also forwarding the error ... to be honest, that function can just be:

```js
pyimport(mod_name: string): Promise<PyProxy> {
  return this._remote.pyimport(mod_name);
}
```

and the `await pyimport(...)` would always do exactly the same with even less ticks around but practically, defining a method `async` alrady makes the returned promise explicit, even if one returns just `1` because the `then` dance you described is looked internally regardless (at the `Number.prototype.then`) so that awaiting upfront without handling a try catch or without doing anything extra is usually considered redundant ... same as:

```js
async function fetchStuff() {
  // do something ...
  return await fetch(...);
  // same as return fetch(...)
}
```

Anyway, minor gotcha, just wanted to point out awaiting without handling triggers errors in the wrong spot and it doesn't bring in anything more, so whenever possible I personally prefer just returning the promise and let the stack do its job in the right place.

Reply:
@hoodmane about this:

> the thing awaited here is a task not a promise

if the task is a primitive then I agree the former is the correct way to handle ... I assumed some object would've been returned and if that's the case with or without `await` is basically the same ... anyway, resolving this as it was never really a blocker, thanks for clarifications.

Reply:
No I just checked and you are absolutely right @WebReflection:
```js
async function f(){
	await sleep(1000);
    console.log("print hi!");
}

function lazyf(){
    return { then(res){ res(f()); } };
}

// g1 is the basically the same as f!
async function g1(){
   return lazyf();
}

// Also the same as g1
function g2(){
   return Promise.resolve(lazyf());
}
```
I think this is a bit counter intuitive...